### PR TITLE
New proposals

### PR DESCRIPTION
**Pull request by dmvass**
Hi @myarik, i would like propose new enhancement:

1. **ElasticAPIView** as base view class for all elasticsearch views.
2. Simplify **ListElasticMixin** class,  because mixin should consist minimal logic and being very simple.
3. Upgrade **ElasticOrderingFilter** for more options customization.
4. Added method **get_es_query** in **ElasticSearchFilter** for simple search query customization.
5. Added new filter validators, because method **clean_field** was very hard to extend.
6. Renamed **es_ordering** to **es_ordering_fields**.

Current API changed by minimum in this updates.

### Some proposals in future:

- Test coverage
- Separate **example_project** to another repo and documented in README file.

### CODE REVIEW

File: rest_framework_elasticsearch/es_filters.py [hunk 1]
```diff
@@ -45,25 +50,34 @@ def get_schema_fields(self, view):
 
 class ElasticOrderingFilter(filters.OrderingFilter, BaseEsFilterBackend):
 
-    def get_default_ordering(self, view):
-        ordering = getattr(view, 'es_ordering', None)
+    def get_es_ordering_fields(self, view):
+        ordering = view.get_es_ordering_fields() or self.ordering_fields
```
Comment (myarik) [inline rest_framework_elasticsearch/es_filters.py]: I think we can leave only a method:  `ordering = view.get_es_ordering_fields()`

File: rest_framework_elasticsearch/es_filters.py [hunk 2]
```diff
@@ -45,25 +50,34 @@ def get_schema_fields(self, view):
...
-        else:
-            valid_fields = [
-                (item, item) if isinstance(item, six.string_types) else item
-                for item in valid_fields
-                ]
-        return valid_fields
+    def get_valid_fields(self, queryset, view, context={}):
+        fields = self.get_es_ordering_fields(view)
+        if not fields:
+            return tuple()
```
Comment (myarik) [inline rest_framework_elasticsearch/es_filters.py]: By default we allow filtering on serializer fields, that is why we need the context value
`return self.get_default_valid_fields(queryset, view, context)`

File: rest_framework_elasticsearch/es_filters.py [hunk 3]
```diff
@@ -147,7 +132,7 @@ def get_schema_fields(self, view):
 class ElasticFieldsRangeFilter(ElasticFieldsFilter):
 
     def get_es_filter_fields(self, view):
-        return getattr(view, 'es_range_filter_fields', tuple())
+        return view.get_es_range_filter_fields() or tuple()
```
Comment (myarik) [inline rest_framework_elasticsearch/es_filters.py]: I think default value need return the method `get_es_range_filter_fields `

**Review by myarik — COMMENTED** [review]
Add small comments

**Comment by dmvass** [comment]
Thanks @myarik! I'm did this fixes.

**Pull request merged**